client/service_create.go

```diff
@@ -62,9 +62,11 @@ func imageWithDigestString(image string, dgst digest.Digest) string {
 	if err == nil {
 		if _, isCanonical := ref.(reference.Canonical); !isCanonical {
 			namedRef, _ := ref.(reference.Named)
```

Comment:
Unrelated to the fix, but I don't think it's okay to assume that this type assertion will succed. `ParseAnyReference` can return something that doesn't satisfy `Named`. Would it work to use `ParseNormalizedNamed` instead?